USING_WITH_PYRIGHT.md

```diff
@@ -0,0 +1,305 @@
+# Pyright and Pylance Together
+
+[Pyright](https://github.com/Microsoft/pyright) is an open source version of a Python type checker that forms the basis for Pylance. Both Pyright and Pylance can be used inside of an IDE to provide a [language server](https://code.visualstudio.com/api/language-extensions/language-server-extension-guide#why-language-server) for Python code.
+
+Pyright can also be used on the [command line](https://github.com/microsoft/pyright/blob/main/docs/command-line.md) to produce json or text output. This can be used in a variety of situations, like a pre-commit hook, or a [github action](https://github.com/jakebailey/pyright-action).
+
+However, Pylance and Pyright don't always produce the same results. The reason for this is a difference in configuration. 
+
+## Stub differences
+
+Pylance comes bundled with a bunch of stubs, usually found here in the installation:
+
+```
+<pylance-extension-install>/dist/bundled
+<pylance-extension-install>/dist/native-stubs
+```
+
+In order for Pyright to produce the same errors as Pylance it need to use the same stub files. 
+
+This might be done by copying the contents of each of those folders to a new folder. 
+
+You'd then reference that new folder in your [pyrightconfig.json](https://microsoft.github.io/pyright/#/configuration)
+
+```json
+{
+    "stubPath": "./pylanceStubs"
+}
+```
+
+Where `pylanceStubs` contains the contents of the `bundled` and `native-stubs` folders.
+
+We have plans to make this easier in the [future](https://github.com/microsoft/pylance-release/discussions/3638).
+
+## Settings differences
+
+Both Pylance and Pyright have [settings](https://github.com/microsoft/pylance-release#settings-and-customization) to control behavior. For the most part, they use the same values, but the few differences do have an impact on the analysis.
+
+| Setting | Pylance default | Pyright default | Description | Potential Impact |
+|----|----|----|----|----|
+| python.analysis.autoSearchPaths | true | false|  Adds 'src' to the list of search paths | This may change what files are found when analyzing. So if you're getting missing imports for things in your 'src' tree, this might be why |
+| python.analysis.indexing | true | false | All files in the current virtual environment are parsed to a certain depth. This gives Pylance more information for auto import completions and other things. | This can have an impact on completions and hover information, but shouldn't impact errors returned.
+| python.analysis.useLibraryCodeForTypes | true | false | When true, this means 3rd party libraries are analyzed to produce type information. Without it, all 3 party libraries are just assumed to be of type Any unless they provided type information explicitly | This setting changes the types found by Pylance/Pyright. It can definitely cause differences in errors reported, so make this the same for both if you want consistent results. A lot of the time this is the sole cause of differences. |

```

Comment:
I've been considering changing pyright's default for `useLibraryCodeForTypes` to true. This would be a breaking change for pyright users, but this would eliminate a common source of confusion for those who want to use both pyright and pylance. Thoughts?

Reply:
I assume it makes analysis slower? Like a lot slower? But more accurate. I'd pick accuracy over speed.

Reply:
I'm not so concerned about analysis speed. The pyright CLI is much less sensitive to speed than pylance because the former is used in a batch manner. There's no human impatiently waiting for immediate feedback. :) 

The issue is with accuracy of type information. The original thinking was that pyright needed accurate type information for type checking, whereas pylance could get away with somewhat inaccurate (inferred) type information because it was focused mostly on completion suggestions. But our type inference has gotten good enough now that I think it's reasonable to default `useLibraryCodeForTypes` to true for pyright. Users can always disable it if it's giving them too many inaccurate types.